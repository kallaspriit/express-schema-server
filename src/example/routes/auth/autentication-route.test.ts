import * as supertest from 'supertest';
import setupApp from '../../app';

let app: supertest.SuperTest<supertest.Test>;

describe('authentication-route', () => {
	beforeEach(async () => {
		app = supertest(await setupApp());
	});

	it('should return logged in user message', async () => {
		const response = await app
			.get('/auth/login')
			.auth('jack', 'daniels')
			.send();

		expect(response.status).toEqual(200);
		expect(response.body.success).toBe(true);
		expect(response.body.payload).toMatchSnapshot();
	});
});
