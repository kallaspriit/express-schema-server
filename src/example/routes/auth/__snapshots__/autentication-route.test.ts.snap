// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authentication-route should return logged in user message 1`] = `
Object {
  "message": "Hello Jack Daniels",
}
`;
